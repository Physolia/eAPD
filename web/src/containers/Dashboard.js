import PropTypes from 'prop-types';
import React from 'react';
import { connect } from 'react-redux';
import { getIsFederal } from '../reducers/user.selector';
import FederalDashboard from '../pages/dashboard/fed-dashboard/FederalDashboard';
import StateDashboard from '../pages/dashboard/state-dashboard/StateDashboard';

const Dashboard = ({ isFederal, ...rest }) =>
  isFederal ? <FederalDashboard {...rest} /> : <StateDashboard {...rest} />;

Dashboard.propTypes = {
  isFederal: PropTypes.bool.isRequired
};

const mapStateToProps = state => ({
  isFederal: getIsFederal(state)
});

export default connect(mapStateToProps)(Dashboard);

export { Dashboard as plain, mapStateToProps };
