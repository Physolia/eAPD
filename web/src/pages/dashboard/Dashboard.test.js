import { shallow } from 'enzyme';
import React from 'react';

import { plain as Dashboard, mapStateToProps } from './Dashboard';

describe('general dashboard component', () => {
  test('renders correctly if not a federal user', () => {
    const component = shallow(<Dashboard isFederal={false} />);
    expect(component).toMatchSnapshot();
  });

  test('renders correctly if federal user', () => {
    const component = shallow(<Dashboard isFederal />);
    expect(component).toMatchSnapshot();
  });

  test('maps state to props', () => {
    expect(
      mapStateToProps({
        user: {
          data: {
            state: {
              id: 'fd'
            }
          }
        }
      })
    ).toEqual({ isFederal: true });

    expect(
      mapStateToProps({
        user: {
          data: {
            state: {
              id: 'fl'
            }
          }
        }
      })
    ).toEqual({ isFederal: false });
  });
});
