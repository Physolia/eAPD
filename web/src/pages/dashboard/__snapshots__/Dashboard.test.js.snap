// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`general dashboard component renders correctly if federal user 1`] = `<Connect(FederalDashboard) />`;

exports[`general dashboard component renders correctly if not a federal user 1`] = `<Connect(StateDashboard) />`;
